src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -309,7 +309,9 @@ public void setPeriod(String period) {
     @Override
     public void visitJavadocToken(DetailNode ast) {
         final Optional<DetailNode> inlineTag = getInlineTagNode(ast);
-        if (inlineTag.isPresent() && isSummaryTag(inlineTag.get())) {
+        if (inlineTag.isPresent()
+            && isSummaryTag(inlineTag.get())
+            && isDefinedFirst(inlineTag.get())) {
```

Comment:
I suggest to refactor this a bit. Instead of checking for node presense and invoking `get()` a million times we can use java11 `Optional#ifPresentOrElse`
in our case it will look like
```
inlineTag.ifPresentOrElse(
    node -> hereIsExtractedMethodForNonNullableNode(node),
    () -> validateUntaggedSummary(ast)
)
```

?

Reply:
Yes, `get()` was called too many times, I extracted it to a variable.
The reason I didn't use `ifPresentOrElse` earlier was due to the logic, the current logic allows us to go to the last else if branch even if the inline tag is present, which `ifPresentOrElse` doesn't allow.